_posts/2020-05-16-adding-typescript-types-github-graphql-api.md

```diff
@@ -136,42 +137,45 @@ Great, we're all done with the wizard. Install all the plugins the wizard wrote
 npm install
 ```
 
-Our last setup step is to write the `src/generated/github-schema-loader.ts` file we referenced earlier. Create a file at
+Now, write the `src/generated/github-schema-loader.ts` file we referenced earlier. Create a file at
 `src/generated/github-schema-loader.ts` and paste the following code:
 
 ```ts
-import { schema } from '@octokit/graphql-schema'
+import { schema } from "@octokit/graphql-schema";
 export default schema.json;
 ```
 
 This will load the schema up from the package published by Github,
 [`@octokit/graphql-schema`](https://github.com/octokit/graphql-schema).
 
-Finally, you'll need to add `ts-node/register` to your `codegen.yml` file so the `github-schema-loader.ts` can be
+Finally, you'll need to add `ts-node/register` to your `codegen.ts` file so the `github-schema-loader.ts` can be
 transpiled. The file should look like this:
 
-```yml
-# codegen.yml
-overwrite: true
-schema: "src/generated/github-schema-loader.ts"
-documents: null
-generates:
-  src/generated/graphql.ts:
-    plugins:
-      - "typescript"
-      - "typescript-resolvers"
-      - "typescript-document-nodes"
-# Add this block
-require:
-  - ts-node/register
+```ts
+// codegen.ts
+import type { CodegenConfig } from "@graphql-codegen/cli";
+
+const config: CodegenConfig = {
+  overwrite: true,
+  schema: "src/generated/github-schema-loader.ts",
+  generates: {
+    "src/generated/graphql.ts": {
+      plugins: ["typescript", "typescript-resolvers", "typescript-document-nodes"],
+    },
+  },
+  // Add this line
+  require: ["ts-node/register"],
```

Comment:
@Hacksore I think the ESM issues you were running into were caused by not adding this line in your config.

I see that, in your sample repo, you made your project an ESM project, which is another solution, too. I kept the blog post referencing ts-node, since it should work for all projects.